bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -171,6 +171,11 @@ type AdditionalUserData struct {
 	// +optional
 	Config string `json:"config,omitempty"`
 
+	// Data allows to pass arbitrary set of key/value pairs consistent with
+	// https://cloudinit.readthedocs.io/en/latest/reference/modules.html
+	// to extend existing cloud-init configuration
+	Data map[string]string `json:"data,omitempty"`
```

Comment:
We should probably add something to say that **Data** and **Config** cannot both be used.

Reply:
Added a kubebuilder validation annotation